lib/pbench/cli/server/report.py

```diff
@@ -56,18 +92,38 @@ class Watch:
     """
 
     def __init__(self, interval: float):
-        self.interval = datetime.timedelta(seconds=interval) if interval else None
-        self.start = datetime.datetime.now()
-        self.last = self.start
+        """Initialize the object.
+
+        Args:
+            interval: interval in seconds for status updates
+        """
+        self.start = time.time()
+        self.interval = interval
+        self.status = "starting"
+        if interval:
+            self.thread = Thread(target=self.watcher)
+            self.thread.setDaemon(True)
+            self.thread.start()
 
     def update(self, status: str):
-        now = datetime.datetime.now()
-        if self.interval and now > self.last + self.interval:
-            self.last = now
-            delta = now - self.start
-            hours, remainder = divmod(delta.seconds, 3600)
+        """Update status if appropriate.
+
+        If periodic updates are enabled, and {interval} time has elapsed since
+        the last update, print the update message.
```

Comment:
I recommend something more like
> Updates the message which will be printed at the next interval if periodic updates are enabled.

Reply:
Yeah, I neglected to update the description. Sigh.